source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -56,6 +56,21 @@ class ImmediateMutationChecker {
   std::unique_ptr<Checker> rule_checker_;
 };
 
+void ExtProcStreamStats::recordGrpcCallStats(
+    std::chrono::microseconds latency, Grpc::Status::GrpcStatus call_status,
+    ProcessorState::CallbackState callback_state,
+    envoy::config::core::v3::TrafficDirection traffic_direction) {
+  ASSERT(callback_state != ProcessorState::CallbackState::Idle);
+  grpcStats(traffic_direction).stats_.emplace_back(latency, call_status, callback_state);
+}
+
+ExtProcStreamStats::GrpcStats&
+ExtProcStreamStats::grpcStats(envoy::config::core::v3::TrafficDirection traffic_direction) {
+  return traffic_direction == envoy::config::core::v3::TrafficDirection::INBOUND
```

Comment:
ASSERT that it's not unspecified perhaps?

Reply:
Thanks, assertion added.